src/http-client/index.ts

```diff
@@ -1,68 +1,14 @@
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
```

Comment:
I ran into circular dependency issues, so that's why I broke out the type definitions to a separate module.